packages/libs/error-stack/src/fmt/hook.rs

```diff
@@ -189,8 +265,237 @@ impl<T> HookContext<'_, T> {
     ///
     /// [`alternate()`]: Self::alternate
     /// [`Debug`]: core::fmt::Debug
-    pub fn attach_snippet(&mut self, snippet: impl Into<String>) {
-        self.parent.snippets.push(snippet.into());
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # // we only test with Rust 1.65, which means that `render()` is unused on earlier version
+    /// # #![cfg_attr(not(rust_1_65), allow(dead_code, unused_variables, unused_imports))]
+    /// use std::io::ErrorKind;
+    ///
+    /// use error_stack::Report;
+    ///
+    /// struct Error {
+    ///     code: usize,
+    ///     reason: &'static str,
+    /// }
+    ///
+    /// Report::install_debug_hook::<Error>(|Error { code, reason }, ctx| {
+    ///     if ctx.alternate() {
+    ///         ctx.snippet(format!("Error {code}:\n  {reason}"));
+    ///     }
```

Comment:
Same as above. We probably only want to show the `.alternate()` once, but I don't really mind either way.

Reply:
fixed in [7640562](https://github.com/hashintel/hash/pull/1008/commits/76405629a6351a476bb7a4be3a1591fdc26ebca2)